fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/domain/SearchQuery.java

```diff
@@ -71,13 +71,18 @@ public void add(SearchExtension ext) {
      */
     public <T> T visit(SearchQueryVisitor<T> visitor) throws FHIRPersistenceException {
         logger.entering(CLASSNAME, "visit");
-        T query = getRoot(visitor);
 
-        // Pre-process any extensions before we process the parameters
+        // Get the root query and process extensions for it
+        T query = getRoot(visitor);
         visitExtensions(query, visitor);
 
+        // Add the parameters subquery and add the extensions there too
         T parameterBase = visitor.getParameterBaseQuery(query);
+        for (SearchExtension ext: this.extensions) {
+            ext.visit(parameterBase, visitor);
+        }
         visitSearchParams(parameterBase, visitor);
+
```

Comment:
Any other files ? or just these two?

Reply:
just these two.  basically just the 3-line addition on 81-83.  the hard/scary part is ensuring it doesn't break anything.  it passed all the tests locally and so it might just come down to ensuring it doesn't have unexpected negative perf impact on any other queries outside of the one I was focused on